src/redfish/messages/messages.py

```diff
@@ -0,0 +1,174 @@
+#! /usr/bin/python
+# Copyright Notice:
+# Copyright 2019-2020 DMTF. All rights reserved.
+# License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/Redfish-Tacklebox/blob/main/LICENSE.md
+
+"""
+Messages Module
+
+File : messages.py
+
+Brief : This file contains the definitions and functionalities for interacting
+        with Messages for a given Redfish service
+"""
+import re
+
+class RedfishOperationFailedError( Exception ):
+    """
+    Raised when an operation has failed (HTTP Status >= 400)
+    """
+    pass
+
+class RedfishPasswordChangeRequiredError( Exception ):
+    """
+    Raised when password change required
+    """
+    def __str__(self):
+        return  "\n{}\nURL: {}\n".format( str(self.args[0]), str(self.args[1]) )
+
+def print_error_payload( response ):
+    """
+    Prints an error payload, which can also be used for action responses
+
+    Args:
+        response: The response to print
+    """
+
+    try:
+        print( get_error_messages( response ) )
+    except:
+        # No response body
+        if response.status >= 400:
+            print( "Failed" )
+        else:
+            print( "Success" )
+
+def get_messages_detail( response ):
+    """
+    Builds messages detail dict in the payload
+
+    Args:
+        response: The response to parser
+
+    Returns:
+        The dict containing messages_detail
+        messages_detail["status"]: http status code
+        messages_detail["successful"]: response successful (http status code < 400)
+        messages_detail["code"]: redfish message response code field
+        messages_detail["@Message.ExtendedInfo"]: redfish message response code field
+    """
+
+    messages_detail = {}
+    messages_detail["status"] = response.status
+    messages_detail["successful"] = False
+    messages_detail["code"] = ""
```

Comment:
Specifically "code" and "message" at the root of an error object have confused users, and the Redfish spec recommends clients ignoring this. This is because the `@Message.ExtendedInfo` property is expected to contain every message necessary for the client to consume.

I recommend structuring the "message_detail" dictionary like this to ease of use:

```
messages_detail = {
    "Status": response.status,
    "Successful": true/false,
    "Messages": [ <copy of the contents of @Message.ExtendedInfo ]
}
```

We could also have a fallback where if `@Message.ExtendedInfo` is not present, we can form the object like this:

```
messages_detail = {
    "Status": response.status,
    "Successful": true/false,
    "Messages": [
        { "MessageId": response.dict["error"]["code"], "Message": response.dict["error"]["message"] }
    ]
}
```